File: FE/WEB/app/models/otp.js
import Model, { attr, belongsTo } from '@ember-data/model';

export default class OtpModel extends Model {
  //@belongsTo('tenant', { async: true, inverse: null }) tenant;

  @attr('string', { defaultValue: '' }) userId;
  @attr('string', { defaultValue: '' }) tenantId;
  @attr('string', { defaultValue: '' }) otpValue;
  @attr('string', { defaultValue: '' }) otpSended;
  @attr('date') creationDate;
  @attr('boolean', { defaultValue: true }) isValid;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/Otp.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;
using MIT.Fwk.Core.Attributes;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [Table("Otps")]
    [SkipJwtAuthentication(JwtHttpMethod.GET)]
    public class Otp : Identifiable<string>
    {
        [Attr]
        public string UserId { get; set; }

        [Attr]
        public int TenantId { get; set; }

        [Attr]
        public string OtpValue { get; set; }

        [Attr]
        public string OtpSended { get; set; }

        [Attr]
        public DateTimeOffset? CreationDate { get; set; }

        [Attr]
        public bool IsValid { get; set; }

        [HasOne]
        public virtual User User { get; set; }

        [HasOne]
        public virtual Tenant Tenant { get; set; }
    }
}
